### DISCUSSION TITLE
converting data from Kraken exchange

### DISCUSSION THREAD

**Discussion by PythonNewby** [DaveSkender/Stock.Indicators#1077] [Help and support]
Hi,

Could someone help with converting Kraken-Data into the standard OHLCV Iterable[Quote]format?

This is what I use to have my data from the Kraken exchange:

resp = requests.get('https://api.kraken.com/0/public/OHLC?pair=SOLUSD&amp;since=1647625329&amp;interval=60')

Could someone suggest how to transform the data I get with this method into the Quote format which then can be used with the tools on this site?

Many thanks!

**Comment by Nikos-Salichos** [comment]
I can help you regarding C#.
You can create your own class in C#, deserialize the JSON response to your class and then use the following LINQ to map it to quotes.

```csharp
var quotes = yourData 
    .Select(q=> new Quote { 
        Open = q.OpenPrice, 
        High = q.HighPrice, 
        Low = q.LowPrice, 
        Close = q.ClosePrice, 
        Date = q.CloseDate
    })
    .OrderBy(q => q.Date)
    .ToList();
```

Later you can for example loop like:
`foreach (var indicatorResult in quotes.GetStoch()) { … }`

and indicatorResult will contain all the output data.

I am not author or contributor, I have just implement the current library for my own (personal - non commercial) project. Feel free to take a look:
https://github.com/Nikos-Salichos/TechnicalAnalysis
